File: Editor/src/main/java/org/turbomedia/turboedit/editor/windows/PreferencesWindow.java
package org.turbomedia.turboedit.editor.windows;

import javafx.scene.Scene;
import javafx.stage.Modality;
import javafx.stage.Stage;
import org.turbomedia.turboedit.editor.panes.PreferencesPane;

import static org.turbomedia.turboedit.editor.Editor.ICON;
import static org.turbomedia.turboedit.editor.Editor.TITLE;
import static org.turbomedia.turboedit.editor.misc.Locale.GetText;

public class PreferencesWindow extends Stage {

    public PreferencesWindow() {
        setTitle(TITLE + " - " + GetText("title.preferences"));
        getIcons().add(ICON);

        final int WIDTH = 600;
        setMinWidth(WIDTH);
        setWidth(WIDTH);

        final int HEIGHT = 600;
        setMinHeight(HEIGHT);
        setHeight(HEIGHT);

        setScene(new Scene(new PreferencesPane()));

        initModality(Modality.APPLICATION_MODAL);
        showAndWait();
    }

}

File: Editor/src/main/kotlin/turboedit/editor/windows/preferences/PreferencesWindow.kt
package turboedit.editor.windows.preferences

import javafx.scene.Scene
import javafx.stage.Modality
import javafx.stage.Stage
import turboedit.editor.Editor
import turboedit.editor.misc.Locale
import turboedit.editor.panes.preferences.PreferencesPane

class PreferencesWindow : Stage() {
    init {
        title = Editor.Companion.TITLE + " - " + Locale.getText("title.preferences")
        icons.add(Editor.Companion.ICON)

        val width = 600
        minWidth = width.toDouble()
        setWidth(width.toDouble())

        val height = 700
        minHeight = height.toDouble()
        setHeight(height.toDouble())

        scene = Scene(PreferencesPane())

        centerOnScreen()
        initModality(Modality.APPLICATION_MODAL)
        showAndWait()
    }
}
